File: fastapi_server/examples/quickstart/config/lifecycles/10_startup_lifecycle.py
"""Lifecycle hooks for fastapi_server."""

import logging

log = logging.getLogger("fastapi_server.lifecycle")


def on_init(app, config):
    log.info("init hook: boot complete (title=%s port=%s)", config.title, config.port)


async def on_startup(app, _config):
    log.info("startup hook: server is ready for traffic")


async def on_shutdown(app, _config):
    log.info("shutdown hook: draining connections")

File: fastify_server/examples/quickstart/config/lifecycles/10_startup.lifecycle.mjs
export function onInit(server, config) {
  server.log.info({ title: config.title, port: config.port }, "init hook: polyglot boot complete");
}

export async function onStartup(server) {
  server.log.info("startup hook: server is ready for traffic");
}

export async function onShutdown(server) {
  server.log.info("shutdown hook: draining connections");
}
